stdlib/asm/crypto/dsa/eddsa/ed25519.masm

```diff
@@ -0,0 +1,298 @@
+use std::crypto::hashes::rpo
+use std::crypto::hashes::sha2_512
+use std::sys
+use std::word
+
+# ENCODING CONVENTIONS
+# ================================================================================================
+#
+# - Byte data is stored in memory as packed u32 values in little-endian format
+# - Each u32 packs four bytes: u32 = u32::from_le_bytes([b0, b1, b2, b3])
+# - Unused bytes in the final u32 must be zero
+# - Memory addresses supplied to this module must be word-aligned (divisible by 4)
+
+# CONSTANTS
+# ================================================================================================
+
+const EDDSA_VERIFY_EVENT = event("stdlib::crypto::dsa::eddsa25519::verify")
+
+const PK_LEN_FELTS = 8        # 32.div_ceil(4)
+const K_DIGEST_LEN_FELTS = 16 # 64.div_ceil(4)
+const SIG_LEN_FELTS = 16      # 64.div_ceil(4)
+
+# PROCEDURES
+# ================================================================================================
+
+#! Verifies an EdDSA (Ed25519) signature with commitment-based public key verification.
+#!
+#! This high-level wrapper mirrors the pattern used in ECDSA verification (`verify_ecdsa_k256_keccak`).
+#! Given a public key commitment and the original message, it reconstructs the data needed for
+#! EdDSA verification. The public key and signature are supplied via the advice stack.
+#!
+#! Inputs:
+#!   Operand stack: [PK_COMM, MSG, ...]
+#!   Advice stack:  [PK[8] | SIG[16] | ...]
+#! Outputs:
+#!   Operand stack: []
+#!   Advice stack:  []
+#!
+#! Where:
+#! - `PK_COMM`: RPO hash commitment of the 32-byte Ed25519 public key
+#! - `MSG`: single word (4 field elements) representing the message to verify
+#! - `PK[8]`: 32-byte public key packed as 8 field elements on advice stack
+#! - `SIG[16]`: 64-byte signature packed as 16 field elements on advice stack
+#!
+#! Local memory layout (element addresses):
+#!   - locaddr[0..8]:   public key (32 bytes packed as 8 felts)
+#!   - locaddr[8..16]:  message (32 bytes = 8 felts)
+#!   - locaddr[16..32]: signature (64 bytes packed as 16 felts)
+#!   - locaddr[32..56]: SHA-512 input buffer (R || PK || MSG = 96 bytes = 24 felts)
+#!   - locaddr[56..72]: k_digest = SHA-512(R || PK || MSG) (64 bytes = 16 felts)
+#!
+#! The procedure traps if:
+#! - The public key does not hash to `PK_COMM` (invalid commitment)
+#! - The signature verification fails
+@locals(72)
+pub proc verify_eddsa_ed25519_sha512
```

Comment:
Question: why not name this just `verify` as we do for ECDSA?

Reply:
I matched ECDSA now, thanks for flagging